src/main/java/mServer/crawler/sender/arte/ArteJsonObjectToDatenFilmCallable.java

```diff
@@ -82,51 +85,9 @@ public DatenFilm call() {
-                                .url(videosUrlVideoDetails2).build();
-                        
-                        try(Response responseVideoDetails2 = httpClient.newCall(request).execute())
-                        {
-                          if(responseVideoDetails2.isSuccessful())
-                          {
-                            /*
-                             * Grobe Struktur des Json's:
-                             * {
-                             *  "meta": {}
-                             *  "programs": [{                   (JSON_ELEMENT_BROADCAST_ELTERNKNOTEN_1)
-                                  "broadcastProgrammings": [{    (JSON_ELEMENT_BROADCAST_ELTERNKNOTEN_2)
-                                    "broadcastBeginRounded": "2016-07-06T02:40:00Z",     (JSON_ELEMENT_BROADCAST)
-                                  ]}
-                                ]}
-                             * }
-                             */
-                            JsonObject jsonObjectVideoDetails2 = gson.fromJson(responseVideoDetails2.body().string(), JsonObject.class);
-                            if(jsonObjectVideoDetails2.isJsonObject() && 
-                                jsonObjectVideoDetails2.get(JSON_ELEMENT_BROADCAST_ELTERNKNOTEN_1).getAsJsonArray().size() > 0 &&
-                                jsonObjectVideoDetails2.get(JSON_ELEMENT_BROADCAST_ELTERNKNOTEN_1).getAsJsonArray().get(0).getAsJsonObject()
-                                  .get(JSON_ELEMENT_BROADCAST_ELTERNKNOTEN_2).getAsJsonArray().size() > 0 &&
-                                jsonObjectVideoDetails2.get(JSON_ELEMENT_BROADCAST_ELTERNKNOTEN_1).getAsJsonArray().get(0).getAsJsonObject()
-                                  .get(JSON_ELEMENT_BROADCAST_ELTERNKNOTEN_2).getAsJsonArray().get(0).getAsJsonObject()
-                                  .has(JSON_ELEMENT_BROADCAST)) 
-                            {
-                                JsonElement jsonBegin = jsonObjectVideoDetails2
-                                    .get(JSON_ELEMENT_BROADCAST_ELTERNKNOTEN_1).getAsJsonArray().get(0).getAsJsonObject()
-                                    .get(JSON_ELEMENT_BROADCAST_ELTERNKNOTEN_2).getAsJsonArray().get(0).getAsJsonObject()
-                                    .get(JSON_ELEMENT_BROADCAST);
-                                if(jsonBegin != JsonNull.INSTANCE) {
-                                    broadcastBegin = jsonBegin.getAsString();    
-                                }
-                            }
-                          }
-                        }
-
-                        film = createFilm(thema, urlWeb, titel, video, broadcastBegin, durationAsTime, beschreibung);
+                        ArteVideoDetailsDTO details = getVideoDetails(gson, programId);                      
+
+                        film = createFilm(thema, urlWeb, titel, video, details.getBroadcastBegin(), durationAsTime, beschreibung);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "details" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)